src/gurt/telemetry.c

```diff
@@ -315,6 +344,9 @@ d_tm_free_node(uint64_t *shmem_root, struct d_tm_node_t *node)
 	if (node == NULL)
 		return;
 
+	if (!d_tm_serialization_enabled)
+		return;
```

Comment:
I guess I don't follow here... We don't free nodes at all unless we've been initialized as the client?

Reply:
`We don't free nodes at all unless we've been initialized as the client?`

That's correct.  d_tm_free_node() is designed to free the resources of a node.  The only resource within a node is the mutex.  The server producer process will no longer allocate mutexes for serialization, so there is nothing to free on a per node basis if serialization is disabled.  When the producer process exits, it calls d_tm_fini().  This function has the comments relating to the mutex objects and the call d_tm_free_node(), but does not call this function.   Given that we don't call it, it's a candidate for being deleted.  I left it in for now for flexibility.

Prior to this PR, I disabled calling d_tm_free_node() (and it remains disabled) because it would have been incorrect to destroy the mutexes upon server shutdown because they were shared with the client.  The d_tm_free_node() has even less utility after this PR, but I left it in for future reference as a basis for cleaning up the shared mutex objects that will be created on behalf of a client application producer process, should we ever use the API for client application side telemetry.

Reply:
Thanks for the clarification! LGTM.